src/binding.cc

```diff
@@ -3571,3 +3574,73 @@ void v8__PropertyDescriptor__set_configurable(v8::PropertyDescriptor* self,
 }
 
 }  // extern "C"
+
+// cppgc
+
+extern "C" {
+
+using RustTraceFn = void (*)(cppgc::Visitor*, void* obj);
+using RustDestroyFn = void (*)(void* obj);
+
+class RustObj final: public cppgc::GarbageCollected<RustObj> {
+  public:
+    explicit RustObj(void* obj, RustTraceFn trace, RustDestroyFn destroy): trace_(trace), destroy_(destroy), obj_(obj) {}
+
+    ~RustObj() {
+      destroy_(obj_);
+    }
+
+    void Trace(cppgc::Visitor* visitor) const {
+      trace_(visitor, obj_);
+    }
```

Comment:
I'd recommend swapping the order of these parameters as this may allow us to inline this function into a vtable directly.